cassiopeia/data/CassiopeiaTree.py

```diff
@@ -147,6 +147,10 @@ def populate_tree(self, tree: Union[str, ete3.Tree, nx.DiGraph]):
                 self.__network.nodes[u]["time"] + self.__network[u][v]["length"]
             )
 
+    def __check_network_initialized(self):
+        if self.__network is None:
+            raise CassiopeiaTreeError("Tree has not been initialized.")
+
```

Comment:
I really like this -- one of the things I was thinking about doing last night, per your suggestion, was add a decorator to make these checks. I couldn't figure out how to do it, but perhaps this would be even cleaner.

I'm happy with this solution though.

Reply:
I figured that (1) a decorator doesn't cut down on the lines of code any more than this (both approaches just use 1 line of code); and (2) decorators are hackier (whereas the current code can be understood by any python novice). So my conclusion was: the problem I am trying to solve here is so simple that using a decorator is overkill.

Reply:
Sounds good -- so pragmatic of you! :-). I'm good to go on this PR whenever Richard has an opportunity to review it.